### PR TITLE
Replace FILO linked list with FIFO ringbuffer

### PR DESCRIPTION
**Pull request by CurlyMoo**
A linked list is (without additional measures) a first in last out data structure. That's not what we want for buffering commands. Therefor the linked list is replaced by a ringbuffer which is the most efficiënt first in first out data structure.

This ringbuffer is also faster than the previous linked list and prevents unnecessary memory fragmentation.

The only downside is that we need to claim (129 * buffersize) bytes of memory at front.

---

This PR is the second PR of a series working towards implementing rules.

### CODE REVIEW

**Comment by IgorYbema** [comment]
I was thinking about this while implementing it. I couldn't find a reason why FILO was not good enough instead of FIFO. I'll take a look

**Comment by CurlyMoo** [comment]
> I couldn't find a reason why FILO was not good enough instead of FIFO. I'll take a look

That we can't think of it att front, doesn't mean those issues aren't going to pop-up. I think users should be able to assume commands are done in order. E.g.:
Turn HP on -> Z1_Heat_Curve_Target_High_Temp -> Z2_Heat_Curve_Target_High_Temp -> Turn HP off

Is very different from the same commands in reverse order:

Turn HP off -> Z1_Heat_Curve_Target_High_Temp -> Z2_Heat_Curve_Target_High_Temp -> Turn HP on

The first results in the heatpump being off, the second in it being on.

File: HeishaMon/HeishaMon.ino [hunk 1]
```diff
@@ -230,29 +232,23 @@ bool readSerial()
...
-      newCommand->value[i] = command[i];
-    }
-    newCommand->next = commandBuffer;
-    commandBuffer = newCommand;
-    commandsInBuffer++;
-  }
-  else {
-    log_message((char*)"Too much commands already in buffer. Ignoring this commands.");
+  if(cmdnrel+1 > MAXCOMMANDSINBUFFER) {
+    log_message((char *), "Too much commands already in buffer. Ignoring this commands.\n");
```
Comment (IgorYbema) [inline HeishaMon/HeishaMon.ino:245]: The comma after (char *) must be removed

Comment (CurlyMoo) [reply HeishaMon/HeishaMon.ino]: Fixed

**Comment by IgorYbema** [comment]
tested ok

**Pull request merged**